src/core/SampleBuffer.cpp

```diff
@@ -675,6 +681,38 @@ f_cnt_t SampleBuffer::decodeSampleOGGVorbis(
 #endif // LMMS_HAVE_OGGVORBIS
 
 
+#ifdef LMMS_HAVE_MP3IMPORT
+f_cnt_t SampleBuffer::decodeSampleMP3(
+	QString fileName,
+	sample_t * & buf,
+	ch_cnt_t & channels,
+	sample_rate_t & samplerate
+)
+{
+	SNDFILE *sndfile;
+	SF_INFO sfinfo;
+	memset(&sfinfo, 0, sizeof(sfinfo));
+
+	sndfile = sf_open(fileName.toStdString().c_str(), SFM_READ, &sfinfo);
+	if (!sndfile)
+	{
+		printf("Error opening MP3 file.\n");
+		return 0;
+	}
+
+	f_cnt_t frames = sfinfo.frames;
+	channels = sfinfo.channels;
+	samplerate = sfinfo.samplerate;
+
+	buf = new sample_t[frames * channels];
+	frames = sf_readf_float(sndfile, buf, frames);
+
+	sf_close(sndfile);
+
+	return frames;
+}
+#endif
+
```

Comment:
```suggestion
```
Since you are using sndfile, this shouldn't be too different from ``decodeSampleSF``, which does the same thing. If you want to prevent decoding MP3's if the support is not available, I think it should be done in that function.